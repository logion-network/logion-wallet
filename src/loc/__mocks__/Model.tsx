import {
    rejectProtectionRequest,
    acceptProtectionRequest,
} from './ModelMock';

export {
    rejectProtectionRequest,
    acceptProtectionRequest,
};
