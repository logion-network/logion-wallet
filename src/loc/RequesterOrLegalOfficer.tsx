import { LocData, LegalOfficer } from "@logion/client";
import { LocType, UUID } from "@logion/node-api";
import { Col, OverlayTrigger, Tooltip } from "react-bootstrap";
import Ellipsis from "src/common/Ellipsis";
import Icon from "src/common/Icon";
import NewTabLink from "src/common/NewTabLink";
import LocItemDetail from "./LocItemDetail";
import { Viewer } from "./types";

export interface Props {
    loc: LocData;
    viewer: Viewer;
    detailsPath: (locId: UUID, type: LocType) => string;
    legalOfficer?: LegalOfficer;
}

export default function RequesterOrLegalOfficer(props: Props) {
    const { loc, viewer, detailsPath, legalOfficer } = props;

    return (
        <Col md={ 4 } className="closed-icon-container">
            {
                viewer === "LegalOfficer" &&
                <LocItemDetail
                    label="Requested by"
                >
                    { loc.userIdentity?.firstName || "" } { loc.userIdentity?.lastName || "" }
                    {
                        loc.requesterAddress !== null && loc.requesterAddress !== undefined &&
                        <>
                            <OverlayTrigger
                                placement="top"
                                delay={ 500 }
                                overlay={
                                    <Tooltip
                                        id={ loc.requesterAddress }>{ loc.requesterAddress }</Tooltip> }>
                                <span><br /> { loc.requesterAddress }</span>
                            </OverlayTrigger>
                        </>
                    }
                    {
                        loc.requesterLocId !== undefined &&
                        <span><br />
                        <NewTabLink
                            href={ detailsPath(loc.requesterLocId, 'Identity') }
                            iconId="loc-link"
                            inline
                        >
                            <Ellipsis
                                maxWidth="250px">{ loc.requesterLocId.toDecimalString() }</Ellipsis>
                        </NewTabLink>
                    </span>
                    }
                </LocItemDetail>
            }
            {
                viewer === "User" &&
                <LocItemDetail
                    label="Legal Officer in charge"
                >
                    { legalOfficer?.name || "" }
                        <OverlayTrigger
                            placement="top"
                            delay={ 500 }
                            overlay={
                                <Tooltip
                                    id={ loc?.ownerAddress }>{ loc?.ownerAddress }</Tooltip> }>
                            <span><br /> { loc?.ownerAddress }</span>
                        </OverlayTrigger>
                </LocItemDetail>
            }
            {
                loc.closed && loc.voidInfo === undefined &&
                <div className="closed-icon">
                    <Icon icon={ { id: "polkadot_shield" } } />
                </div>
            }
            {
                loc.voidInfo !== undefined &&
                <div className="closed-icon">
                    <Icon icon={ { id: "void_shield" } } />
                </div>
            }
        </Col>
    );
}
