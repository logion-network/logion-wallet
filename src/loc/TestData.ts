import { LegalOfficerCase, UUID, Hash } from "@logion/node-api";
import { LocData, HashString } from "@logion/client";

export function buildLocRequest(locId: UUID, loc: LegalOfficerCase): LocData {
    return {
        ownerAddress: loc.owner,
        requesterAddress: loc.requesterAddress,
        requesterLocId: loc.requesterLocId,
        description: "Description",
        locType: loc.locType,
        createdOn: "",
        decisionOn: "",
        id: locId,
        status: loc.closed ? "CLOSED" : "OPEN",
        files: loc.files.map((locFile, index) => ({
            ...locFile,
            nature: HashString.fromValue(`File ${index}`),
            name: `File ${index}`,
            addedOn: "",
            published: false,
            restrictedDelivery: false,
            contentType: "text/plain",
            size: 42n,
            status: "DRAFT",
        })),
        metadata: loc.metadata.map((locFile, index) => ({
            ...locFile,
            name: HashString.fromValue(`Data ${index}`),
            value: HashString.fromValue(`Value ${index}`),
            addedOn: "",
            published: false,
            status: "DRAFT",
        })),
        links: loc.links.map((locFile, index) => ({
            ...locFile,
            name: `Link ${index}`,
            nature: HashString.fromValue(`Nature ${index}`),
            addedOn: "",
            target: locFile.id.toString(),
            published: false,
            status: "DRAFT",
        })),
        closed: false,
        verifiedIssuer: false,
        issuers: [],
    };
}
