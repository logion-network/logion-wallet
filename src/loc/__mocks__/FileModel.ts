export { getAllDeliveries, getAllCollectionDeliveries } from './FileModelMock';
