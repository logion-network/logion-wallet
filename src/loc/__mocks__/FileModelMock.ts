export let getAllDeliveries = jest.fn().mockResolvedValue({});
