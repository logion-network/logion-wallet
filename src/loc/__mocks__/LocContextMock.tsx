import { LocData, CollectionItem } from "@logion/client";
import { UUID, ValidAccountId } from "@logion/node-api";

import { LocItem } from "../LocItem";

let locState: any = {
    isLogionIdentity: () => false,
    isLogionData: () => true,
};

export function setLocState(value: any) {
    locState = value;
    if(value.data) {
        loc = {
            ...loc,
            ...locState.data(),
        }
    }
}

let locItems: LocItem[] = [];

export function setLocItems(items: LocItem[]) {
    locItems = items;
}

let locId = new UUID("aed4c6e4-979e-48ad-be6e-4bd39fb94762");

export function setLocId(id: UUID) {
    locId = id;
}

let loc: Partial<LocData> | null = {
    id: locId,
    locType: "Identity",
    requesterAccountId: ValidAccountId.polkadot("5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW"),
    userIdentity: {
        firstName: "John",
        lastName: "Doe",
        email: "john.doe@logion.network",
        phoneNumber: "+1234",
    }
};

export function setLocRequest(request: Partial<LocData> | null) {
    loc = request;
}

export let refresh = jest.fn();

export let requestSof = jest.fn();

export function setRequestSof(fn: jest.Mock) {
    requestSof = fn;
}

export let deleteMetadata = jest.fn();

export let deleteLink = jest.fn();

let collectionItems: CollectionItem[] = [];

export function setCollectionItems(items: CollectionItem[]) {
    collectionItems = items;
}

async function mutateLocState(mutator: (state: any) => Promise<any>): Promise<void> {
    await mutator(locState);
}

export function useLocContext() {
    return {
        linkLoc: {

        },
        locId,
        loc,
        locItems,
        refresh,
        requestSof,
        deleteMetadata,
        deleteLink,
        checkResult: { result: "NONE" },
        locState,
        collectionItems,
        mutateLocState,
    };
}
