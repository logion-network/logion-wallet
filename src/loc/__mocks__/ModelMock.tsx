export let rejectRequest = jest.fn();

export let acceptRequest = jest.fn();

export function setAcceptRequest(mockFn: any) {
    acceptRequest = mockFn;
}

export let setAssetDescription = jest.fn();

export let rejectProtectionRequest = jest.fn();

export function setRejectProtectionRequest(mockFn: any) {
    rejectProtectionRequest = mockFn;
}

export let acceptProtectionRequest = jest.fn();

export function setAcceptProtectionRequest(mockFn: any) {
    acceptProtectionRequest = mockFn;
}

export function decision(legalOfficerAddress: string, decisions: any[]) {
    return decisions[0];
}

export let fetchRecoveryInfo = jest.fn();

export function setFetchRecoveryInfo(mockFn: any) {
    fetchRecoveryInfo = mockFn;
}

export let acceptLocRequest = jest.fn();

export function setAcceptLocRequest(mockFn: any) {
    acceptLocRequest = mockFn;
}

export let rejectLocRequest = jest.fn();

export function setRejectLocRequest(mockFn: any) {
    rejectLocRequest = mockFn;
}

export let addLink = jest.fn();
