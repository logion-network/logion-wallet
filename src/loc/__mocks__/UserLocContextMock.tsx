import { UUID } from "@logion/node-api/dist/UUID";
import { LocData } from "@logion/client/dist/Loc";

import { LocRequest } from "../../common/types/ModelTypes";
import { LocItem } from "../types";

let locState: any;

export function setLocState(value: any) {
    locState = value;
}

export let refresh = jest.fn();

let requestSof = jest.fn();

export function setRequestSof(fn: jest.Mock) {
    requestSof = fn;
}

let requestSofOnCollection = jest.fn();

export function setRequestSofOnCollection(fn: jest.Mock) {
    requestSofOnCollection = fn;
}

let locId: UUID;

export function setLocId(id: UUID) {
    locId = id;
}

let loc: LocData | null = null;

export function setLoc(request: LocData | null) {
    loc = request;
}

let locItems: LocItem[] = [];

export function setLocItems(items: LocItem[]) {
    locItems = items;
}

export let deleteMetadata = jest.fn();

export let deleteFile = jest.fn();

export let addFile = jest.fn().mockResolvedValue(undefined);

export let addMetadata = jest.fn().mockResolvedValue(undefined);

export function useUserLocContext() {

    return {
        refresh,
        locState,
        requestSof,
        requestSofOnCollection,
        locId,
        loc,
        deleteMetadata,
        deleteFile,
        locItems,
        addFile,
        addMetadata,
    }
}
