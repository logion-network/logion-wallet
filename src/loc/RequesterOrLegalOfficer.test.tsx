import { LocData } from "@logion/client";
import { UUID } from "@logion/node-api";
import { PATRICK } from "src/common/TestData";

import { shallowRender } from "src/tests"
import { TEST_WALLET_USER } from "src/wallet-user/TestData";
import { CLOSED_IDENTITY_LOC_ID, DEFAULT_LEGAL_OFFICER } from "src/__mocks__/LogionMock";
import RequesterOrLegalOfficer from "./RequesterOrLegalOfficer";

describe("RequesterOrLegalOfficer", () => {

    it("renders polkadot requester for LO", () => {
        const element = shallowRender(<RequesterOrLegalOfficer
            loc={ locWithPolkadotRequester }
            viewer="LegalOfficer"
            { ...otherProps }
        />);
        expect(element).toMatchSnapshot();
    })

    it("renders logion requester for LO", () => {
        const element = shallowRender(<RequesterOrLegalOfficer
            loc={ locWithLogionRequester }
            viewer="LegalOfficer"
            { ...otherProps }
        />);
        expect(element).toMatchSnapshot();
    })

    it("renders LO in charge for User", () => {
        const element = shallowRender(<RequesterOrLegalOfficer
            loc={ locWithPolkadotRequester }
            viewer="User"
            legalOfficer={ PATRICK }
            { ...otherProps }
        />);
        expect(element).toMatchSnapshot();
    })
})

const locWithPolkadotRequester = {
    ownerAccountId: DEFAULT_LEGAL_OFFICER,
    requesterAccountId: TEST_WALLET_USER,
    requesterLocId: UUID.fromDecimalStringOrThrow(CLOSED_IDENTITY_LOC_ID),
    userIdentity: {
        firstName: "John",
        lastName: "Doe",
    },
    closed: false,
} as unknown as LocData;

const locWithLogionRequester = {
    ownerAccountId: DEFAULT_LEGAL_OFFICER,
    requesterLocId: UUID.fromDecimalStringOrThrow(CLOSED_IDENTITY_LOC_ID),
    userIdentity: {
        firstName: "John",
        lastName: "Doe",
    },
    closed: false,
} as unknown as LocData;

const otherProps = {
    detailsPath: () => "",
};
