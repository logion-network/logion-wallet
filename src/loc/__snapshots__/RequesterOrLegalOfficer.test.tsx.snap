// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequesterOrLegalOfficer renders LO in charge for User 1`] = `
<div
  className="closed-icon-container"
>
  <LocItemDetail
    label="Legal Officer in charge"
  >
    Patrick Gielen
    <OverlayTrigger
      defaultShow={false}
      delay={500}
      overlay={
        <Tooltip
          id="vQx5kESPn8dWyX4KxMCKqUyCaWUwtui1isX6PVNcZh2Ghjitr"
          placement="right"
        >
          vQx5kESPn8dWyX4KxMCKqUyCaWUwtui1isX6PVNcZh2Ghjitr
        </Tooltip>
      }
      placement="top"
      trigger={
        Array [
          "hover",
          "focus",
        ]
      }
    >
      <span>
        <br />
         
        vQx5kESPn8dWyX4KxMCKqUyCaWUwtui1isX6PVNcZh2Ghjitr
      </span>
    </OverlayTrigger>
  </LocItemDetail>
</div>
`;

exports[`RequesterOrLegalOfficer renders logion requester for LO 1`] = `
<div
  className="closed-icon-container"
>
  <LocItemDetail
    label="Requested by"
  >
    John
     
    Doe
    <span>
      <br />
      <NewTabLink
        href=""
        iconId="loc-link"
        inline={true}
      >
        <Ellipsis
          maxWidth="250px"
        >
          85833363768713528858922097642089825569
        </Ellipsis>
      </NewTabLink>
    </span>
  </LocItemDetail>
</div>
`;

exports[`RequesterOrLegalOfficer renders polkadot requester for LO 1`] = `
<div
  className="closed-icon-container"
>
  <LocItemDetail
    label="Requested by"
  >
    John
     
    Doe
    <React.Fragment>
      <OverlayTrigger
        defaultShow={false}
        delay={500}
        overlay={
          <Tooltip
            id="Polkadot:vQxHAE33LeJYV69GCB4o4YcCgnDu8y99u5hy2751fRdxjX9kz"
            placement="right"
          >
            vQxHAE33LeJYV69GCB4o4YcCgnDu8y99u5hy2751fRdxjX9kz
          </Tooltip>
        }
        placement="top"
        trigger={
          Array [
            "hover",
            "focus",
          ]
        }
      >
        <span>
          <br />
           
          <React.Fragment>
            vQxHAE33LeJYV69GCB4o4YcCgnDu8y99u5hy2751fRdxjX9kz
          </React.Fragment>
        </span>
      </OverlayTrigger>
    </React.Fragment>
  </LocItemDetail>
</div>
`;
